app/models/link.rb

```diff
@@ -87,4 +94,23 @@ def ownership_stake
       "; percent stake: #{stake}%" if stake.present?
     end
   end
+
+  # Tell Rails not to try writing to this model, since it is backed by a view
+  def readonly?
+    true
+  end
+
+  # Refresh the view, concurrently if it has already been populated
+  def self.refresh
```

Comment:
I'm curious, do you know about how long this takes? I can run some tests too on my environment

Reply:
Of course it varies greatly depending on how much refreshing is to be done.

I added some performance tests to the model spec: 7ec69d2

I find these consistently take a few ms per refresh, always below 10 and normally below 5. However, when running in a test environment we of course don't have all the relationship data.

I thus also added a benchmarker which could be run against a full-size relationships table: ec70d9f.

These are of course much slower. Concurrent refreshes are about twice as slow, but they perform better in another sense: they don't lock the view, so other queries can proceed unhindered during the refresh. From benchmarks in my dev environment, you're looking at about 30 seconds between relationship updates and the relevant links being available to queries. Here are my benchmark results:
```

Warming up --------------------------------------
       basic refresh     1.000  i/100ms
  concurrent refresh     1.000  i/100ms
Calculating -------------------------------------
       basic refresh      0.066  (± 0.0%) i/s -      1.000  in  15.073793s
  concurrent refresh      0.031  (± 0.0%) i/s -      1.000  in  31.762955s

Comparison:
       basic refresh:        0.1 i/s
  concurrent refresh:        0.0 i/s - 2.11x  (± 0.00) slower
```

Reply:
To run the same benchmark yourself: `bundle exec ruby script/benchmarks/links_view.rb`

Reply:
I get similar times 17s/30s. 

One change I'll going to do is have `Link.refresh` run in a background job, because HTTP requests right now hang when creating new relationships.

Reply:
Oh yes, it should obviously run in a job. Good catch.